crypto/ec/ecx_meth.c

```diff
@@ -410,6 +397,52 @@ static int ecx_get_pub_key(const EVP_PKEY *pkey, unsigned char *pub,
     return 1;
 }
 
+static size_t ecx_pkey_dirty_cnt(const EVP_PKEY *pkey)
+{
+    /*
+     * We provide no mechanism to "update" an ECX key once it has been set,
+     * therefore we do not have to maintain a dirty count.
+     */
+    return 1;
+}
+
+static void *ecx_pkey_export_to(const EVP_PKEY *pk, EVP_KEYMGMT *keymgmt,
+                                int want_domainparams)
+{
+    const ECX_KEY *key = pk->pkey.ecx;
+    OSSL_PARAM_BLD tmpl;
+    OSSL_PARAM *params;
+    void *provdata = NULL;
+
+    /* There are no domain parameters for ECX keys */
+    if (want_domainparams)
+        return NULL;
+
+    ossl_param_bld_init(&tmpl);
+
+    /* A key must at least have a public part */
```

Comment:
Is this always true if it generates the public part from the private part?

Reply:
Yes. Internally if we have the private key then we have the public key too. The only time that isn't true is when the key is first created or the ECK_KEY is first constructed. But that is part of key gen or key mgmt. (Not yet moved that stuff to providers - another PR)